postrelease-vip.php

```diff
@@ -2,567 +2,680 @@
-	register_deactivation_hook( __FILE__, 'postrelease_notify_deactivate' );	
-}
-
-/**
- * Appends the PostRelease javascript to header 
- */
-function postrelease_wp_enqueue_scripts() {
-	//if plugin is activated and we are not in an admin page
-	if ( 1 == get_option( 'prx_plugin_activated', 0 ) && ! is_admin() ) {
-	    wp_register_script('postrelease', PRX_JAVASCRIPT_URL);
-	    wp_enqueue_script('postrelease');
-	}
-}
-
-function postrelease_init() {
-	$args = array('public' => false || PRX_DEV,
-				  'publicly_queryable' => true,
-	              'label' => 'Sponsored Post',
-	    		  'exclude_from_search' => true,
-	    		  'can_export' => false,	 
-	    		  'supports' => array( 'title', 'editor', 'author', 'thumbnail', 'excerpt' )
-	);
-	register_post_type( PRX_CUSTOM_TYPE, $args );
-}
-
-/**
- * Function when plugin is activated.
- * In WP VIP, the activate hook is not supported.
- * The plugin is activated when the user goes to the dashboard and
- * adds his publication to the PostRelease network. 
- */
-function postrelease_activate() {
-	if ( ! function_exists( 'wpcom_is_vip' ) ) {
-		flush_rewrite_rules(); //not vip
-	} else {
-		do_action( 'postrelease_activate' ); //vip
-	}
-}
+if ( ! class_exists( PostRelease_VIP ) ) :
+
+	class PostRelease_VIP {
+
+		/**
+		 * The one instance of PostRelease_VIP
+		 *
+		 * @var PostRelease_VIP
+		 */
+		private static $instance;
+
+		/**
+		 * Post type
+		 *
+		 * @var string
+		 */
+		public $post_type = 'pr_sponsored_post';
+
+		/**
+		 * Plugin version
+		 *
+		 * @var string
+		 */
+		public $plugin_version = '1.3';
+
+		/**
+		 * DB Version
+		 *
+		 * @var string
+		 */
+		public $db_version = 1;
+
+		/**
+		 * In development mode?
+		 *
+		 * @var string
+		 */
+		public $is_dev = true;
+
+		/**
+		 * URL of postrelease server
+		 *
+		 * @var string
+		 */
+		public $postrelease_server = 'http://www.postrelease.com';
+
+		/**
+		 * URL of postrelease js file
+		 *
+		 * @var string
+		 */
+		public $postrelease_js_url = 'http://a.postrelease.com/serve/load.js?async=true';
+
+		/**
+		 * Initialize the class.
+		 */
+		public function __construct() {}
+
+		/**
+		 * Return an instance of this class.
+		 *
+		 * @return object
+		 */
+		public static function get_instance() {
+			if ( is_null( self::$instance ) ) {
+				self::$instance = new self();
+				self::$instance->setup();
+			}
+			return self::$instance;
+		}
 
-/**
- * Reset changes caused by the plugin 
- */
-function postrelease_deactivate() {
-	//delete all sponsored posts
-	$sponsored_posts_array = get_pages(array('post_type' => PRX_CUSTOM_TYPE));
-	foreach( $sponsored_posts_array as $sponsored_post ) {
-		wp_delete_post( $sponsored_post->ID, true);
-	}
+		/**
+		 * Setup actions and filters, etc.
+		 *
+		 * @return void
+		 */
+		public function setup() {
+			$this->postrelease_server = trailingslashit( $this->postrelease_server );
+
+			// Actions
+			add_action( 'init', [ $this, 'postrelease_init' ] );
```

Comment:
All of these methods are prefixed with `postrelease_`, which is unnecessary and redundant considering that we're within a class. keep it simple.